File: RPC/leo/RPC04/ProblemaC.cpp
#include <iostream>
#include <string>
#include <iterator>
#include <cmath>
using namespace std;

string construccion(int difDia, int difH, int difM){
	string res, difDiaS, difMS, difHS;
	res = "";
	difDiaS  = to_string(difDia);
	difHS  = to_string(difH);
	difMS  = to_string(difM);
	if(difDia == 0 || difDia == -1){
		if(difH == 0){
			res = res + difMS + " minutes";
		}else{
			if(difM == 0){
				res = res + difHS + " hours";
			}else{
				res = res + difHS + " hours and " + difMS + " minutes";
			}
		}
	}else{
		if(difH == 0){
			if(difM == 0){
				res = res + difDiaS + " days";
			}else{
				res = res + difDiaS + " days and " + difMS + " minutes";
			}
		}else{
			if(difM == 0){
				res = res + difDiaS + " days and " + difHS + " hours";
			}else{
				res = res + difDiaS + " days, " + difHS + " hours, " + difMS + " minutes";
			}
		}
	}
	return res;
}

int resta(int m1, int m2){
	int res = 0;
	if(m1 > m2){
		res = 60 - m1 + m2;
	}else{
		res = abs(m1 - m2);
	}
	return res;
}

int restaH(int h1, int h2){
	int res = 0;
	if(h1 > h2){
		res = 24 - h1 + h2;
	}else{
		res = abs(h1 - h2);
	}
	return res;
}

int comparacion(string dia1, string dia2, string lista[], int tamLista){
	int res = 0;
	for(int i = 0; i < tamLista; i++){
		if(lista[i] == dia1){
			for(int j = i; j < tamLista; j++){
				if(lista[j] == dia2){
					res = abs(i - j) - 1;
				}
			}
			if(res == 0){
				for(int j = 0; j < tamLista; j++){
					if(lista[j] == dia2){
						j += 7;
						res = abs(i - j) - 1;
					}
				}
			}
		}
	}
	return res;
}

int desgloceM(string cadena){
	string res = "";
	for(int i = 7; i < 9; i++){
		res += cadena.at(i);
	}
	return stoi(res);
}

int desgloceH(string cadena){
	string res = "";
	for(int i = 4; i < 6; i++){
		res += cadena.at(i);
	}
	return stoi(res);
}

string desgloceDia(string cadena){
	string res = "";
	for(int i = 0; i < 3; i ++){
		res += cadena.at(i);
	}
	return res;
}

string ejercicio(string cadena1, string cadena2){
	string dia1, dia2;
	int difDia, difH, difM, hora1, hora2, min1, min2, tamLista;
	string lista[] = {"Mon", "Tue", "Wed", "Thu", "Fri", "Sat", "Sun"};
	tamLista = size(lista);

	dia1 = desgloceDia(cadena1);	
	dia2 = desgloceDia(cadena2);
	hora1 = desgloceH(cadena1);
	hora2 = desgloceH(cadena2);
	min1 = desgloceM(cadena1);
	min2 = desgloceM(cadena2);
	difDia = comparacion(dia1, dia2, lista, tamLista);
	difH = restaH(hora1, hora2);
	difM = resta(min1, min2);

	return construccion(difDia, difH, difM);
}

int main(){
	string cadena1, cadena2, res;

	getline(cin, cadena1);
	getline(cin, cadena2);

	res = ejercicio(cadena1, cadena2);
	cout << res;
}

File: RPC/jose/RPC04/ProblemaC.java
import java.util.Scanner;
public class ProblemaC{
	public static void main(String[] args){
		Scanner sc = new Scanner(System.in);
		String cad1, cad2, res;
		cad1 = sc.nextLine();
		cad2 = sc.nextLine();
		ProblemaC pC = new ProblemaC();
		res = pC.ejercicio(cad1, cad2);
		System.out.println(res);

	}

	public String ejercicio(String cad1, String cad2){
		String dia1, dia2;
		String list[] = {"Mon", "Tue", "Wed", "Thu", "Fri", "Sat", "Sun"};
		int difDia, difH, difM, hora1, hora2, min1, min2;

		dia1 = desgloceDia(cad1);
		dia2 = desgloceDia(cad2);
		hora1 = desgloceH(cad1);
		hora2 = desgloceH(cad2);
		min1 = desgloceM(cad1);
		min2 = desgloceM(cad2);
		difDia = comparacion(dia1, dia2, list);
		difH = restaH(hora1, hora2);
		difM = resta(min1, min2);
		return contruccion(difDia, difH, difM);
	}

	private String desgloceDia(String cadena){
		String res = "";
		for(int i = 0; i < 3; i++){
			res += cadena.charAt(i);
		}
		return res;
	}

	private int desgloceH(String cadena){
		String res = "";
		int r = 0;
		for(int i = 4; i < 6; i++){
			res += cadena.charAt(i);
		}
		r = Integer.parseInt(res);
		return r;
	}
	private int desgloceM(String cadena){
		String res = "";
		int r = 0;
		for(int i = 7; i < 9; i++){
			res += cadena.charAt(i);
		}
		r = Integer.parseInt(res);
		return r;
	}

	private int comparacion(String dia1, String dia2, String[] lista){
		int res = 0;
		for(int i = 0;  i < lista.length; i++){
			if(lista[i].equals(dia1)){
				for(int j = i; j < lista.length; j++){
					if(lista[j].equals(dia2)){
						res = Math.abs(i - j) - 1;
					}
				} 
				if(res == 0){
					for(int j = 0; j < lista.length; j++){
						if(lista[j].equals(dia2)){
							j += 7;
							res = Math.abs(i - j) - 1;
						}
					}	
				}
			}
		}
		return res;
	}

	private int restaH(int h1, int h2){
		int res = 0;
		if(h1 > h2){
			res = 24 - h1 + h2;
		}else{
			res = h2 - h1;
		}
		return res;
	}

	private int resta(int m1, int m2){
		int res = 0;
		if(m1 > m2){
			res = 60 - m1 + m2;
		}else{
			res = m2 - m1;
		}
		return res;
	}

	private String contruccion(int difDia, int difH, int difM){
		String res = "";
		if(difDia == 0 || difDia == -1){
			if(difH == 0){
				res = res + difM + " minutes";
			}else{
				if(difM == 0){
					res = res + difH + " hours";
				}else{
					res = res + difH + " hours and " + difM + " minutes";
				}
			}
		}else{
			if(difH == 0){
				if(difM == 0){
					res = res + difDia + " days";
				}else{
					res = res + difDia + " days and " + difM + " minutes";
				}
			}else{
				if(difM == 0){
					res = res + difDia + " days and " + difH + " hours";
				}else{
					res = res + difDia + " days, " + difH + " hours, " + difM + " minutes";
				}
			}
		}
		return res;
	}
}